### PR TITLE
FLUID-5489: Documentation for new ChangeApplier features, "includeSource...

### PR DESCRIPTION
**Pull request by amb26**
..." and "excludeSource" as modelListener directives

This pull request contains documentation accompanying https://github.com/fluid-project/infusion/pull/553

### CODE REVIEW

**Comment by amb26** [comment]
This is ready for another round of review

File: src/documents/ChangeApplierAPI.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ A `<modelPathReference>` has the form:
 <table>
     <thead>
         <tr>
-            <th colspan="3">Syntax definition for &#60;modelPathReference&#62; - the key in modelListeners options block for a modelRelayComponent</th>
+            <th colspan="3">Syntax definition for &lt;modelPathReference&gt; - the key in modelListeners options block for a modelRelayComponent</th>
```
Comment (jobara) [inline src/documents/ChangeApplierAPI.md]: It may make sense to codifiy `<modelPathReference>`

File: src/documents/ChangeApplierAPI.md [hunk 2]
```diff
@@ -62,12 +62,17 @@ The 4 examples presented in the _"Examples"_ column are parallel for the two cas
 
 #### Model Listener Declaration ####
 
-A model listener declaration block has exactly the same form and meaning as any of the record types supported by [Invokers](Invokers.md) and [Listeners](EventInjectionAndBoiling.md#listener-boiling) - including the one-string compact syntax documented with [Invokers](Invokers.md). The only difference is that an extra [context name](Contexts.md) is available in this block by the name of change. This is bound to the particular change event which triggered this listener. This context behaves as an object with the following fields:
+A model listener declaration block has exactly the same form and meaning as any of the record types supported by [Invokers](Invokers.md) and [Listeners](EventInjectionAndBoiling.md#listener-boiling) - including the one-string compact syntax documented with [Invokers](Invokers.md). There are two extra features that are supported in
```
Comment (jobara) [inline src/documents/ChangeApplierAPI.md]: Should the second link to "Invokers" be directly to the compact syntax?
https://github.com/amb26/infusion-docs/blob/FLUID-5489/src/documents/Invokers.md#compact-format

File: src/documents/ChangeApplierAPI.md [hunk 3]
```diff
@@ -77,23 +82,104 @@ A model listener declaration block has exactly the same form and meaning as any
...
-            <td>{change}.oldValue</td>
+            <td><code>{change}.oldValue</code></td>
             <td>Any type</td>
             <td>The previous value which was held at the matched model path, before it was overwritten by the change being listened to</td>
         </tr>
         <tr>
-            <td>{change}.path</td>
+            <td><code>{change}.path</code></td>
             <td>String</td>
             <td>The path at which this change occurred. In general this will be the same as the path registered as the modelPathReference for this block - however it may be one segment longer if a wildcard path was used (see next section)</td>
```
Comment (jobara) [inline src/documents/ChangeApplierAPI.md]: Can we provide a link and specific name for "see next section"?

File: src/documents/ChangeApplierAPI.md [hunk 4]
```diff
@@ -77,23 +82,104 @@ A model listener declaration block has exactly the same form and meaning as any
...
+        </tr>
+    </tbody>
+</table>
+
+Example featuring source filtering:
+
+```javascript
+fluid.defaults("examples.sourceExample1", {
+    gradeNames: ["fluid.modelRelayComponent", "autoInit"],
+    model: "initial value",
```
Comment (jobara) [inline src/documents/ChangeApplierAPI.md]: Should this first model actually be 

```
model: {
    things: "initial value"
}
```

**Comment by jobara** [comment]
For the _emphasized **not recommended**_ block under the [Programatic Style](https://github.com/amb26/infusion-docs/blob/FLUID-5489/src/documents/ChangeApplierAPI.md#programmatic-style) section, could we put that as a **Note:** like we do in other documentation?

**Comment by amb26** [comment]
Thanks, @jobara - ready once more

**Comment by jobara** [comment]
merged at 07df1b3d6bcf373efb31540358ecb05b4fa8c308

**Pull request merged**